doc/md/icp-features/1-randomness.md

```diff
@@ -0,0 +1,104 @@
+
+:::info
+Before using the [`fuzz`](https://mops.one/fuzz) or [`idempotency-keys`](https://mops.one/idempotency-keys) packages, ensure that [Mops](https://mops.one/) is installed and initialized in your Motoko project.
+:::
+
+## `raw_rand`
+
+The `raw_rand` function is a system API provided by the [ICP management canister](https://internetcomputer.org/docs/references/system-canisters/management-canister) for requesting cryptographic randomness derived from the network’s verifiable random function. `raw_rand` generates fresh entropy in every execution round, making it suitable for applications requiring high security such as key generation or lotteries where fairness is a legal requirement.
+
+Since `raw_rand` operates asynchronously, canisters must await its response before using the generated bytes. Each call returns a 32-byte (256-bit) random `Blob`.
+
+```motoko no-repl
+actor {
+  let SubnetManager : actor {
+    raw_rand() : async Blob;
+  } = actor "aaaaa-aa";
+
+  public func randomBytes() : async Blob {
+    await SubnetManager.raw_rand();
+  };
+}
+```
+
+## `Random`
+
+The [`Random` module](https://internetcomputer.org/docs/motoko/base/Random) provides an interface that wraps the `raw_rand` function. Since `raw_rand` returns raw bytes, the `Random` module simplifies working with this entropy by offering structured methods for consuming randomness efficiently. The module includes `Random.blob()` for fetching fresh 32-byte entropy and `Random.Finite`, which provides a finite source of randomness that can be used until exhausted. When entropy runs out, a new random `Blob` must be fetched asynchronously.
+
+Below is an example demonstrating how to generate a random boolean using `Random.Finite`.
+
+```motoko no-repl
+import Random "mo:base/Random";
+
+actor {
+  public func randomBoolean() : async ?Bool {
+    let entropy = await Random.blob();
+    let finite = Random.Finite(entropy);
+    // Consumes 1 byte of entropy
+    finite.coin();
+  };
+}
+```
+
+## `fuzz`
+
+The [`fuzz` package](https://mops.one/fuzz) is a random data generator designed primarily for testing, but it can also be used for generating random account IDs, unique values, and randomized inputs. It supports various data types, including numbers, text, arrays, `Blob`s, and principals. The randomness source is customizable, allowing initialization with a time-based seed, a fixed seed for reproducibility, a random `Blob` for stronger entropy, or a custom generator function. By default, `fuzz` uses `Time.now()` as a seed, providing immediate access to pseudo-random values without external dependencies.
+
+The following example demonstrates initialization with the default seed and generating a random `Nat`.
+
+```motoko no-repl
+import Fuzz "mo:fuzz";
```

Comment:
```suggestion
import Fuzz "mo:fuzz/Fuzz";
```
or such... the `mops` package being `fuzz` and the module therein `Fuzz.mo`. But this needs to be double-checked!